### PR TITLE
Reduce complexity http routes

### PR DESCRIPTION
**Pull request by Bond-009**

### CODE REVIEW

**Comment by cvium** [comment]
Is there any point in changing the servicestack source further?

**Comment by Bond-009** [comment]
@cvium is there any reason not to?

**Comment by cvium** [comment]
Well, we should move to Web API asap as the inclusion of ServiceStack is a license violation.

**Comment by scheidleon** [comment]
@cvium ServiceStack shouldn't be a license violation. They have a explicit exception in their license file for FOSS applications.
https://github.com/ServiceStack/ServiceStack/blob/master/license.txt (Line 15)

**Comment by cvium** [comment]
The list does not include GPLv3.

**Comment by Bond-009** [comment]
I think the proper solution is adding a note at the start of the files in question that they should be phased out. But we can still change them when we need to.

File: Emby.Server.Implementations/HttpServer/HttpListenerHost.cs [hunk 1]
```diff
@@ -655,11 +643,6 @@ private void RedirectToSecureUrl(IHttpRequest httpReq, IResponse httpRes, string
             }
             else
             {
-                // TODO what is this?
-                var httpsUrl = url
-                    .Replace("http://", "https://", StringComparison.OrdinalIgnoreCase)
-                    .Replace(":" + _config.Configuration.PublicPort.ToString(CultureInfo.InvariantCulture), ":" + _config.Configuration.PublicHttpsPort.ToString(CultureInfo.InvariantCulture), StringComparison.OrdinalIgnoreCase);
-
```
Comment (JustAMan) [inline Emby.Server.Implementations/HttpServer/HttpListenerHost.cs]: was that broken, or was that just dead code?

Comment (Bond-009) [reply Emby.Server.Implementations/HttpServer/HttpListenerHost.cs]: Dead, the var is never used.

**Pull request merged**